src/applications/ivc-champva/10-10d-extended/helpers/validations.js

```diff
@@ -65,3 +65,143 @@ export const validateOHIDates = (errors, data) => {
+ * @property {boolean} [item.hasMedicarePartD] When `true` and plan supports Part D ('ab' or 'c'), Part D dates and cards are validated.
+ * @property {string} [item.medicarePartDEffectiveDate] Required past date when Part D is present and supported.
+ * @property {string} [item.medicarePartDTerminationDate] Optional; if provided, must be a past date when Part D is present and supported.
+ * @property {{ medicarePartAEffectiveDate?: string }} [item['view:medicarePartAEffectiveDate']] Container for Part A effective date (must be a past date when required).
+ * @property {{ medicarePartBEffectiveDate?: string }} [item['view:medicarePartBEffectiveDate']] Container for Part B effective date (must be a past date when required).
+ * @property {Array} [item.medicarePartAFrontCard] Required uploaded file array for Part A front card.
+ * @property {Array} [item.medicarePartABackCard] Required uploaded file array for Part A back card.
+ * @property {Array} [item.medicarePartBFrontCard] Required uploaded file array for Part B front card.
+ * @property {Array} [item.medicarePartBBackCard] Required uploaded file array for Part B back card.
+ * @property {Array} [item.medicarePartAPartBFrontCard] Required uploaded file array for Parts A & B front card.
+ * @property {Array} [item.medicarePartAPartBBackCard] Required uploaded file array for Parts A & B back card.
+ * @property {Array} [item.medicarePartCFrontCard] Required uploaded file array for Part C front card.
+ * @property {Array} [item.medicarePartCBackCard] Required uploaded file array for Part C back card.
+ * @property {Array} [item.medicarePartDFrontCard] Required uploaded file array for Part D front card when Part D is present.
+ * @property {Array} [item.medicarePartDBackCard] Required uploaded file array for Part D back card when Part D is present.
+ *
+ * @returns {boolean} `true` if validation fails (invalid/missing fields); `false` if the item is valid.
+ */
+export const validateMedicarePlan = (item = {}) => {
+  const {
+    medicarePlanType,
+    medicarePartCCarrier,
+    medicarePartCEffectiveDate,
+    hasMedicarePartD,
+    medicarePartDEffectiveDate,
+    medicarePartDTerminationDate,
+    medicarePartAFrontCard,
+    medicarePartABackCard,
+    medicarePartBFrontCard,
+    medicarePartBBackCard,
+    medicarePartAPartBFrontCard,
+    medicarePartAPartBBackCard,
+    medicarePartCFrontCard,
+    medicarePartCBackCard,
+    medicarePartDFrontCard,
+    medicarePartDBackCard,
+  } = item;
+  const medicarePartAEffectiveDate =
+    item['view:medicarePartAEffectiveDate']?.medicarePartAEffectiveDate;
+  const medicarePartBEffectiveDate =
+    item['view:medicarePartBEffectiveDate']?.medicarePartBEffectiveDate;
+
+  const isValidPastDate = dateString => {
+    if (!dateString) return false;
+    const date = new Date(dateString);
+    return isValid(date) && !isAfter(date, new Date());
```

Comment:
Would `isValid(date) && isBefore` work here instead of `!isAfter`? Reading negation always stumps me for a sec